File: IdeaProjects/ScalaTutorial/ScalaMessageReader/src/main/scala/scalaMessageReader/util/MessageSender.scala
package scalaMessageReader.util

import java.io.File

import org.springframework.messaging.MessageChannel
import org.springframework.messaging.support.GenericMessage

import scalaMessageReader.MessageReaderConstants

class MessageSender {

  def sendMessages(messageChannel : MessageChannel, filePath : String): Unit = {
    val directory = new File(MessageReaderConstants.FULL_FILE_PATH).listFiles()
    assert(directory != null)
    for (file <- directory) messageChannel.send(new GenericMessage[File](file))
  }
}

File: IdeaProjects/MessageReader/src/main/java/messageReader/util/MessageSender.java
package messageReader.util;

import org.springframework.messaging.MessageChannel;
import org.springframework.messaging.support.GenericMessage;

import java.io.File;

public class MessageSender {

    public MessageSender() {
    }

    public void sendMessages(MessageChannel messageChannel, String filePath) throws Exception {
        File[] files = new File(filePath).listFiles();
        for (File f : files) {
            messageChannel.send(new GenericMessage<>(f));
        }
    }
}
